driver-core/src/main/com/mongodb/operation/AsyncQueryBatchCursor.java

```diff
@@ -50,18 +51,14 @@
     private final int limit;
     private final Decoder<T> decoder;
     private final long maxTimeMS;
-    private volatile AsyncConnectionSource connectionSource;
+    private final Object lock = new Object();
```

Comment:
Though see below, as there still may be a need for a lock to coordinate access to the connectionSource and the cursor.